packages/transformers/js/src/dependency_collector.rs

```diff
@@ -375,10 +373,31 @@ impl<'a> Fold for DependencyCollector<'a> {
 
       // Track the returned require call to be replaced with a promise chain.
       self.require_node = Some(call.clone());
-      return call;
+      call
+    } else if kind == DependencyKind::Require {
+      // Don't continue traversing so that the `require` isn't replaced with undefined
+      node
+    } else {
+      node.fold_children_with(self)
     }
+  }
 
-    return node.fold_children_with(self);
+  fn fold_unary_expr(&mut self, node: ast::UnaryExpr) -> ast::UnaryExpr {
+    // Don't traverse `typeof require` further to `require` with undefined
+    if let ast::UnaryExpr {
+      op: ast::UnaryOp::TypeOf,
+      arg,
+      ..
+    } = &node
+    {
+      if let ast::Expr::Ident(ast::Ident { sym, .. }) = &**arg {
+        if sym == &js_word!("require") && !self.decls.contains(&(sym.clone(), node.span.ctxt())) {
+          return node;
+        }
+      }
+    }
+
+    node.fold_children_with(self)
   }
 
```

Comment:
Seems like `fold_member_expr` is missing.
All passes touching identifier should implement it and fold `prop` only if `computed` is true.

Reply:
Seems like a cause of https://github.com/parcel-bundler/parcel/issues/6323

Reply:
I was wrong. Sorry for the mistake.